Runtime/Shapes/ShapeComponent.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+
+namespace UnityEngine.ProBuilder
+{
+    [RequireComponent(typeof(ProBuilderMesh))]
+    public class ShapeComponent : MonoBehaviour
+    {
+        [SerializeReference]
+        public Shape shape = new Cube();
```

Comment:
there should not be any public fields in classes - they need to be exposed through properties